src/daft-plan/src/planner.rs

```diff
@@ -76,18 +74,103 @@ pub fn plan(logical_plan: &LogicalPlan) -> DaftResult<PhysicalPlan> {
+                            Mean(e) => Mean(e.alias(field_id.id.clone()).into()),
+                            Min(e) => Min(e.alias(field_id.id.clone()).into()),
+                            Max(e) => Max(e.alias(field_id.id.clone()).into()),
+                            List(e) => List(e.alias(field_id.id.clone()).into()),
+                            Concat(e) => Concat(e.alias(field_id.id.clone()).into()),
+                        })
+                        .collect();
+
+                    let second_stage_aggs: Vec<daft_dsl::AggExpr> = intermediate_names
+                        .iter()
+                        .zip(schema.fields.keys())
+                        .zip(aggregations.iter())
+                        .map(|((field_id, original_name), agg_expr)| match agg_expr {
+                            Count(_) => Count(
+                                daft_dsl::Expr::Column(field_id.id.clone().into())
+                                    .alias(&**original_name)
+                                    .into(),
+                            ),
+                            Sum(_) => Sum(daft_dsl::Expr::Column(field_id.id.clone().into())
+                                .alias(&**original_name)
+                                .into()),
+                            Mean(_) => Mean(
+                                daft_dsl::Expr::Column(field_id.id.clone().into())
+                                    .alias(&**original_name)
+                                    .into(),
+                            ),
+                            Min(_) => Min(daft_dsl::Expr::Column(field_id.id.clone().into())
+                                .alias(&**original_name)
+                                .into()),
+                            Max(_) => Max(daft_dsl::Expr::Column(field_id.id.clone().into())
+                                .alias(&**original_name)
+                                .into()),
+                            List(_) => List(
+                                daft_dsl::Expr::Column(field_id.id.clone().into())
+                                    .alias(&**original_name)
+                                    .into(),
+                            ),
+                            Concat(_) => Concat(
+                                daft_dsl::Expr::Column(field_id.id.clone().into())
+                                    .alias(&**original_name)
+                                    .into(),
+                            ),
+                        })
+                        .collect();
+
+                    let result_plan = PhysicalPlan::Aggregate(Aggregate::new(
+                        result_plan.into(),
+                        first_stage_aggs,
+                        vec![],
+                    ));
+                    let result_plan = PhysicalPlan::Coalesce(Coalesce::new(
+                        result_plan.into(),
+                        logical_plan.partition_spec().num_partitions,
```

Comment:
Nit: We already have this value in `num_input_partitions`.
```suggestion
                        num_input_partitions,
```